### PR TITLE
all: use errgo.

### PR DESCRIPTION
**Pull request by rogpeppe**
This almost entirely an automated change, made by running
the errfix comment. (code.google.com/p/rog-go/exp/cmd/errfix)

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/35/

File: internal/router/fieldinclude.go [hunk 1]
```diff
@@ -61,9 +62,10 @@ func (h *fieldIncludeHandler) Handle(hs []BulkIncludeHandler, id *charm.URL, pat
 	// Make the single query.
 	doc, err := h.query(id, selector)
 	if err != nil {
-		return nil, err
+		// Call all the handlers with the resulting query document.
+		return nil, errgo.Mask(err)
 	}
-	// Call all the handlers with the resulting query document.
```
Comment (rogpeppe) [inline internal/router/fieldinclude.go:66]: hmm, this looks wrong. fixing.

File: internal/charmstore/store.go [hunk 2]
```diff
@@ -52,15 +53,15 @@ func (s *Store) AddCharm(url *charm.URL, c charm.Charm) error {
...
 	}
 	defer archive.Close()
 	blobHash, size, err := s.putArchive(archive)
 	if err != nil {
-		return err
+		// Add charm metadata to the entities collection.
+		return errgo.Mask(err)
 	}
 
-	// Add charm metadata to the entities collection.
```
Comment (rogpeppe) [inline internal/charmstore/store.go:63]: as does this.

File: internal/router/fieldinclude.go [hunk 3]
```diff
@@ -61,9 +62,10 @@ func (h *fieldIncludeHandler) Handle(hs []BulkIncludeHandler, id *charm.URL, pat
 	// Make the single query.
 	doc, err := h.query(id, selector)
 	if err != nil {
-		return nil, err
+		// Call all the handlers with the resulting query document.
```
Comment (frankban) [inline internal/router/fieldinclude.go]: This comment seemed better placed before.

File: internal/router/router.go [hunk 4]
```diff
@@ -133,20 +133,21 @@ func (r *Router) Handlers() *Handlers {
 // serveIds serves requests that may be rooted at a charm or bundle id.
 func (r *Router) serveIds(w http.ResponseWriter, req *http.Request) error {
 	if err := req.ParseForm(); err != nil {
-		return err
+		// We can ignore a trailing / because we do not return any
```
Comment (frankban) [inline internal/router/router.go]: Ditto.

Comment (rogpeppe) [inline internal/router/router.go]: done

**Comment by jrwren** [comment]
LGTM

File: internal/router/fieldinclude.go (hunk 1 above)
Comment (rogpeppe) [inline internal/router/fieldinclude.go:66]: done.

File: internal/charmstore/store.go (hunk 2 above)
Comment (rogpeppe) [inline internal/charmstore/store.go:63]: done.

**Comment by frankban** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/36/

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**